src/bridge/token_bridge.cairo

```diff
@@ -399,8 +402,37 @@ pub mod TokenBridge {
         let mut calldata = ArrayTrait::new();
         let dispatcher = IERC20MetadataDispatcher { contract_address: token };
         token.serialize(ref calldata);
-        dispatcher.name().serialize(ref calldata);
-        dispatcher.symbol().serialize(ref calldata);
+
+        // Openzeppelin erc20 used felt252 as return types for `name()` and `symbol()` while
+        // `ByteArray`
+        // is used currently.
+        // So we use underlying syscalls, to support both the interface.
+        // The returned span is deserialized into to a ByteArray in both cases to make it consistent
+        // In case of ByteArray the length of returned span will be at least 3, while exactly 1 for
+        // felt252
+        let name_selector = selector!("name");
+        let mut name = call_contract_syscall(token, name_selector, array![].span())
+            .unwrap_syscall();
+        if (name.len() == 1) {
+            let name_deserialised = name[0].format_as_byte_array(10);
+            name_deserialised.serialize(ref calldata);
+        } else {
+            let name_deserialised = Serde::<ByteArray>::deserialize(ref name).unwrap();
+            name_deserialised.serialize(ref calldata);
+        }
```

Comment:
can be a function inside this function itself so that it can be reused twice?

Reply:
Cairo does not support function inside function, can create a separate function